src/inventory_client/inventory_client.go

```diff
@@ -41,20 +46,64 @@ type EnabledHostData struct {
 	Host      *models.Host
 }
 
-func CreateInventoryClient(clusterId string, inventoryURL string, pullSecret string, logger *logrus.Logger) (*inventoryClient, error) {
+func CreateInventoryClient(clusterId string, inventoryURL string, pullSecret string, insecure bool, caPath string, logger *logrus.Logger) (*inventoryClient, error) {
 	clientConfig := client.Config{}
 	var err error
 	clientConfig.URL, err = url.ParseRequestURI(createUrl(inventoryURL))
 	if err != nil {
 		return nil, err
 	}
 
-	clientConfig.Transport = requestid.Transport(http.DefaultTransport)
+	var certs *x509.CertPool
+	if insecure {
+		logger.Warn("Certificate verification is turned off. This is not recommended in production environments")
+	} else {
+		certs, err = readCACertificate(caPath)
```

Comment:
why is it required? If it's CA authority u dont need cret, right?
I mean while working with api.openshift u will pass nothing

Reply:
If it's a valid public CA you won't use this option. It's only if you use a self-signed CA and still want verification (most on-premised environments to my knowledge).

Reply:
right... but...